ingestion-beam/src/main/java/com/mozilla/telemetry/Decoder.java

```diff
@@ -72,11 +72,14 @@ public static PipelineResult run(DecoderOptions.Parsed options) {
             .apply("ParseUri", ParseUri.of()).failuresTo(failureCollections))
 
         // Special case: decryption of Pioneer payloads
+        // Both legacy messages sent to the `telemetry.pioneer-study` doctype
+        // and the glean.js-styled messages are supported respectively
```

Comment:
Is there an assumption here that only pioneer and rally messages will be flowing through this deployment of the decoder? If `getPioneerEnabled()` is set, will we throw exceptions for doctypes that are not encrypted? Perhaps that's worth mentioning here.

Reply:
Yes and yes, I'll document some of that here.